src/transforms/route.rs

```diff
@@ -87,6 +91,14 @@ pub struct RouteConfig {
     #[configurable(metadata(docs::additional_props_description = "An individual route."))]
     #[configurable(metadata(docs::examples = "route_examples()"))]
     route: IndexMap<String, AnyCondition>,
+
+    /// Flag for exclusive routing.
+    ///
+    /// When this flag is enabled, an event will be routed to only the first matching route.
```

Comment:
Missed this [comment](https://github.com/vectordotdev/vector/issues/17040#issuecomment-1664723707) when I was doing this work. IndexMap should preserve the insertion order but there are macros and parsing steps in the middle. Explicitly ranking the routes might be cleaner.

Reply:
Yeah, that was my main concern about doing it this way when it[was raised before: that YAML/JSON/TOML maps are unordered. In practice, we use `IndexMap` which should preserve the order, but users may be generating Vector configs where the ordering wouldn't be preserved.

My thoughts on ways we could do this are here: https://github.com/vectordotdev/vector/issues/17040#issuecomment-1664723707

Reply:
This is a highly requested feature, though, so happy to see progress on it 🎉

Reply:
Thanks Jesse. Agreed, this is a potential footgun. I will see if it's possible to convert the IndexMap to an array without breaking existing configs. Otherwise, assigning weights is a good backwards compatible alternative.